src/test/java/bio/terra/service/snapshotbuilder/SnapshotAccessRequestModelTest.java

```diff
@@ -15,47 +17,77 @@
 import bio.terra.model.SnapshotBuilderProgramDataRangeCriteria;
 import java.time.Instant;
 import java.util.List;
+import java.util.Map;
 import java.util.UUID;
 import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
 
 @Tag(Unit.TAG)
 class SnapshotAccessRequestModelTest {
   private static final String EXPECTED_LIST_SUMMARY_STRING =
-      "The following concepts from Race: 0, 1, 2";
+      String.format(
+          "The following concepts from Race: %s, %s",
+          SnapshotBuilderTestData.RACE_PROGRAM_DATA_LIST_ITEM_ONE.getName(),
+          SnapshotBuilderTestData.RACE_PROGRAM_DATA_LIST_ITEM_TWO.getName());
   private static final String EXPECTED_RANGE_SUMMARY_STRING = "Year of birth between 1960 and 1980";
-  private static final String EXPECTED_DOMAIN_SUMMARY_STRING = "Condition Concept Id: 401";
+  private static final String EXPECTED_DOMAIN_SUMMARY_STRING = "Condition Concept: name 401";
+  public static final int CONCEPT_ID = 401;
+  private static final Map<Integer, String> conceptIdsToNames = Map.of(CONCEPT_ID, "name 401");
 
   @Test
   void toApiResponse() {
     SnapshotAccessRequestModel model = generateSnapshotAccessRequestModel();
-    compareModelAndResponseFields(model, model.toApiResponse(SnapshotBuilderTestData.SETTINGS));
+    compareModelAndResponseFields(model, model.toApiResponse());
+  }
+
+  @Test
+  void toApiDetails() {
+    SnapshotAccessRequestModel model = generateSnapshotAccessRequestModel();
+    String conditionConceptName = "Condition Concept Name";
+    String expectedSummaryString =
+        "Participants included:\nName: cohort\nGroups:\nMust meet all of:\nThe following concepts from Race: \nCondition Concept: "
+            + conditionConceptName
+            + "\nYear of birth between 1950 and 2000\nTables included:Drug, Condition\n";
+    assertThat(
+        model
+            .generateModelDetails(
+                SnapshotBuilderTestData.SETTINGS,
+                Map.of(SNAPSHOT_BUILDER_COHORT_CONDITION_CONCEPT_ID, conditionConceptName))
```

Comment:
Instead of using a constant for ID you could derive it from the model, now that we have an API to extract concept IDs.
```suggestion
                Map.of(model.generateConceptIds().get(0), conditionConceptName))
```